bin/machinery

```diff
@@ -20,6 +20,12 @@
 require_relative '../lib/machinery'
 
 begin
+  if Dir.exist?(File.join(Machinery::ROOT, ".git"))
+    Dir.chdir(File.join(Machinery::ROOT, "machinery-helper")) do
+      Cheetah.run("rake")
```

Comment:
The check for file changes it quite powerful, it is only run on developer machines and it only takes 0,1 seconds to run.

Running rake also prevents all the methods getting parsed and kept in memory while regular machinery is run.